go/integration_test/processor/transfer_test.go

```diff
@@ -0,0 +1,218 @@
+				}
+				if balance := transactionContext.GetBalance(receiver0); balance.Cmp(value0) != 0 {
+					t.Errorf("receiver balance was not updated correctly, want %v, got %v", value0, balance)
+				}
+				if balance := transactionContext.GetBalance(receiver1); balance.Cmp(value1) != 0 {
+					t.Errorf("receiver balance was not updated correctly, want %v, got %v", value1, balance)
+				}
+			})
+		}
+	}
+}
+
+func TestProcessor_CallsWithInsufficientBalanceAreHandledCorrectly(t *testing.T) {
+	for processorName, processor := range getProcessors() {
+		for callName, call := range callTypesAndProperties() {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+				sender0 := tosca.Address{1}
+				receiver0 := tosca.Address{2}
+				receiver1 := tosca.Address{3}
+				checkValue := byte(55)
+				value := tosca.NewValue(42)
+				receiverBalance := tosca.NewValue(24)
+
+				code0 := pushCallArguments(call, sufficientGas, value, receiver1)
+				code0 = append(code0, []byte{
+					byte(call.callType),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				code1 := []byte{
+					byte(vm.PUSH1), checkValue,
+					byte(vm.PUSH1), byte(0),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}
+
+				state := WorldState{
+					sender0:   Account{},
+					receiver0: Account{Balance: receiverBalance, Code: code0},
+					receiver1: Account{Code: code1},
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender0,
+					Recipient: &receiver0,
+					GasLimit:  sufficientGas,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Fatalf("execution was not successful or failed with error %v", err)
+				}
+
+				// static call and delegate call do not transfer value,
+				// therefore the transaction does not fail with insufficient balance
+				if call.callType == vm.STATICCALL || call.callType == vm.DELEGATECALL {
+					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 31), checkValue)) {
+						t.Errorf("transaction has not been handled correctly, code was not executed")
+					}
+				} else {
+					if !slices.Equal(result.Output, bytes.Repeat([]byte{0}, 32)) {
+						t.Errorf("transaction has not been handled correctly, code was executed")
+					}
+				}
+
+				zero := tosca.NewValue(0)
+				if balance := transactionContext.GetBalance(sender0); balance.Cmp(zero) != 0 {
+					t.Errorf("sender balance was not updated correctly, want %v, got %v", zero, balance)
+				}
+				if balance := transactionContext.GetBalance(receiver0); balance.Cmp(receiverBalance) != 0 {
+					t.Errorf("receiver balance was not updated correctly, want %v, got %v", receiverBalance, balance)
```

Comment:
nit: I think this error message is misleading and it through me off in understanding this test ... what you actually check here is that the receiver balance was not modified, right?